src/_pytest/_code/code.py

```diff
@@ -8,7 +8,7 @@
 from types import TracebackType
 from typing import Generic
 from typing import Optional
-from typing import Pattern
+from typing import Pattern  # noqa: F401 (used in type string)
```

Comment:
if you change it to 'Union[str, Pattern]` then flake8 will be happy and you won't need the noqa

I'll be fixing this in `pyflakes` eventually, just haven't gotten around to it quite yet

Reply:
Thanks, did that.